### PR TITLE
add rootDirectory property

### PR DESCRIPTION
**Pull request by mostafakhudair**
Add \Config\Paths::rootDirectory holds project main folders
Optimize bootstrap

### CODE REVIEW

**Comment by MGatner** [comment]
I'm generally not a fan of defining more directories to track. Can you explain what this allows that isn't available by relative paths to existing directory definitions?

File: system/Test/bootstrap.php [hunk 1]
```diff
@@ -33,7 +33,7 @@
 defined('APPPATH')       || define('APPPATH', realpath(rtrim($paths->appDirectory, '\\/ ')) . DIRECTORY_SEPARATOR);
 defined('WRITEPATH')     || define('WRITEPATH', realpath(rtrim($paths->writableDirectory, '\\/ ')) . DIRECTORY_SEPARATOR);
 defined('SYSTEMPATH')    || define('SYSTEMPATH', realpath(rtrim($paths->systemDirectory, '\\/')) . DIRECTORY_SEPARATOR);
-defined('ROOTPATH')      || define('ROOTPATH', realpath(APPPATH . '../') . DIRECTORY_SEPARATOR);
+defined('ROOTPATH')      || define('ROOTPATH', realpath(rtrim($paths->rootDirectory, '\\/ ')) . DIRECTORY_SEPARATOR);
```
Comment (samsonasik) [inline system/Test/bootstrap.php:36]: it is bc break without property_exists check

File: system/bootstrap.php [hunk 2]
```diff
@@ -37,63 +31,45 @@
...
  * @filesource
  */
 
-/*
+use CodeIgniter\CodeIgniter;
+use CodeIgniter\Config\DotEnv;
+use CodeIgniter\Services;
+use Config\App;
+use Config\Autoload;
+use Config\Modules;
```
Comment (samsonasik) [inline system/bootstrap.php:45]: I created PHPStan rule for it at https://github.com/codeigniter4/CodeIgniter4/pull/3786, no need to check manually ;)

**Comment by samsonasik** [comment]
please revert unrelated change to PR stuff, also it is BC break as I mentioned before without property_exists/isset check.

**Pull request closed without merging**